### ISSUE TITLE
Client requires authentication at every launch

### ISSUE THREAD

**Issue by mophilly** [nextcloud/desktop#4281]
### Expected behaviour

After authenticating once with the browser based scheme it should remember the credentials and use that when starting the client the next time.

### Actual behaviour

It always asks for credentials at every launch of the OS.

### Steps to reproduce

1. Start nextcloud client first time
2. Browser opens that requests login
3. Authenticate
4. Close client
5. Open client

### Client configuration

Client version: 3.4.2

Operating system: MacOS Monterey (12.2.1)

OS language: English

### Server configuration

Nextcloud version: 20.0.9
OS: Linux 2.6.32

### Related Issues
Client always asks for authentication #1716
https://github.com/nextcloud/server/issues/17165

**Comment by TomBayne** [comment]
Can confirm this issue exists on client version 3.4.2, server version 20.0.9.
Client OS: Arch Linux (arch-5.16.10-zen)
Server OS: DietPi Linux v8.1.2

**Comment by mophilly** [comment]
Why is this closed without comment or resolution? It seems a simple problem that likely requires only a simple correction.

**Comment by redhell** [comment]
Really annoying.

At still MacOS 12.3.1 with latest nextcloud server/client

**Comment by serjor** [comment]
This is happening to me too

Version 3.4.4, on openSUSE tumbleweed (KDE).

**Comment by sandrock** [comment]
Happening with 5.13.19-2-MANJARO. 

What is this from the log file?  

> [ info nextcloud.sync.credentials.keychainchunk /build/nextcloud-client/src/nextcloud-client/src/libsync/creds/keychainchunk.cpp:346 ]:	Backend unavailable (yet?) Retrying in a few seconds. "The name org.freedesktop.secrets was not provided by any .service files"

**Comment by jahway603** [comment]
I'm experiencing this same issue on the following:

Client version 3.5.1
Two Servers running versions 23.0.4 & 22.2.7

Client OS: Artix Linux (5.17.8-artix1-1)
Server OS: DietPi Linux v8.4.2 (on both servers)

**Comment by roosweb** [comment]
Same here, reauthentification via browser is needed DAILY at Windows 10 laptop with the latest Nextcloud desktop version 3.5.1. So this is not working for us  if Nextcloud does not have a solution for this.

Reauthentification is NOT needed at my iMac (macOS 10.15) with Nextcloud 3.5.1.

**Comment by Hudratronium** [comment]
It seems this problem / bug still has not that impact - or the bug report ist to old that someone is taking a look into it. some of the older threads i found where trying to workaround with generated application passwords for the clients. Sadly this didn't worked out for me either. Makes it hard to "sell" this solution in a professional environment.

**Comment by mophilly** [comment]
> Makes it hard to "sell" this solution in a professional environment.

I agree. My team adopted NextCloud because it was not intrusive for the end user. Now it is intrusive. This inability to logon without human interaction after first authentication causes us unnecessary support requests.

**Comment by BrunoBernardino** [comment]
This started happening to me with 3.5.2 on Linux. I've downgraded to 3.5.1 and it got fixed there. Server version is 24.0.1. I've disabled the automatic update check for now.

**Comment by alexpdp7** [comment]
Can confirm that downgrading to 3.5.1 does fix the issue. I'm on CentOS Stream 9.

**Comment by Hudratronium** [comment]
Sadly i can't confirm this on my windows clients. Came from 3.5.1 but nothing changed. So this seems to be a bogus one...

**Comment by tflidd** [comment]
> This started happening to me with the desktop sync client version 3.5.2 on Linux. I've downgraded to 3.5.1 and the problem went away. Server version is 24.0.1. I've disabled the automatic update check on the client for now.

This is a regression bug. This report here is for an older version. I would consider this to be a different bug from this one here, where this problem consist through many versions. And it seems to be on Linux platforms only. I'd consider opening a separate report for this (if there isn't one already).

**Comment by roosweb** [comment]
Still the same problem with Nextcloud desktop version 3.5.1 + 3.5.2 at WINDOWS 10 laptop :  requires authentication at every (daily) launch !

**Comment by p0kR** [comment]
same on linux using AppImage with version 3.5.2

**Comment by letsfindaway** [comment]
Hi,

> Happening with 5.13.19-2-MANJARO.

and also for me on openSUSE Leap 15.4
 
> What is this from the log file?
> 
> > [ info nextcloud.sync.credentials.keychainchunk /build/nextcloud-client/src/nextcloud-client/src/libsync/creds/keychainchunk.cpp:346 ]:	Backend unavailable (yet?) Retrying in a few seconds. "The name org.freedesktop.secrets was not provided by any .service files"

Have the same messages. It seems that the `org.freedesktop.secrets` service is not running at the time nextcloud client is started, My solution is to delay the autostart of nextcloud by 10 seconds. Seems to work. I changed the desktop entry (mainly the `Exec` line) in `~/.config/autostart/Nextcloud.desktop` to

```ini
[Desktop Entry]
Name=Nextcloud delayed
GenericName=File Synchronizer
Exec=/usr/bin/bash -c "sleep 10;/usr/bin/nextcloud"
Terminal=false
Icon=nextcloud
Categories=Network
Type=Application
StartupNotify=false
X-GNOME-Autostart-enabled=true
```

Since then all boots go without any problem.

**Comment by jokay** [comment]
Can confirm this problem for **Linux AppImage** version **3.5.3**.

**Comment by mkofahl** [comment]
Also for Linux AppImage version 3.5.3 using tls client certifificate.

`[ warning nextcloud.sync.credentials.keychainchunk /home/user/src/libsync/creds/keychainchunk.cpp:360 ]:	Unable to read "***_clientCertificatePEM:https://***/:0" chunk "0" "Unknown error"`

**Comment by jokay** [comment]
Can confirm this problem for **Linux AppImage** version **3.5.4** *(using an app token)*.

**Comment by simaoafonso-pwt** [comment]
Can confirm this remains a problem on Windows, using the App Token functionality.

**Comment by jokay** [comment]
At least no one cares ...

Relates https://github.com/nextcloud/desktop/issues/1292.

**Comment by Hudratronium** [comment]
Seems we are the only ones effected by this - i can't imagine professional users / big organizations as well as all homeusers would just accept this ....

**Comment by BrunoBernardino** [comment]
I believe most people are just keeping this locked in 3.5.1, probably? It seems like it affects enough variety of OS to not be too difficult to reproduce, but I might just be wrong.

**Comment by rekuenkdr** [comment]
Version 3.5.4 (Ubuntu) (Nextcloud-3.5.4-x86_64.AppImage) has the same issue, keep asking for authentication on every launch too.

**Comment by hramberger** [comment]
On Ubuntu 20.04 I have the same issues. The last version that works as expected is 3.5.1 (Nextcloud-3.5.1-x86_64.AppImage). Versions 3.5.2 to 3.5.4 always ask to login at launch. I tested all the mentioned client versions with server version 24.0.4.

**Comment by BrunoBernardino** [comment]
Thanks for confirming, I stopped testing after 2 updates (3.5.2 and 3.5.3) ignored this problem, and there's no confirmation on this from anyone at Nextcloud, here, that I'm aware of.

**Comment by EricDriussi** [comment]
Eeeeey count me in!

**Comment by Hudratronium** [comment]
I am not so sure but seeing that this issue still has the "old" layout for bug-reporting it might be werth to open a new one with the "correct" Tags applied... ?

**Comment by Hudratronium** [comment]
Interesting - another "Issue" with kind of the same effects we are faceing here...

https://github.com/nextcloud/desktop/issues/4796

**Comment by Hudratronium** [comment]
new stable release 3.6 -- Still the same problem...

**Comment by jokay** [comment]
imho, this bug seems reliable / stable enough to be considered as feature now :wink:

**Comment by rekuenkdr** [comment]
I can confirm that Nextcloud-3.6.0-x86_64.AppImage fixes the issue, or at least for me.
Using latest Nextcloud 24.0.5 and Ubuntu 20.04.

**Comment by p0kR** [comment]
Same for me, fixed in Nextcloud-3.6.0-x86_64.AppImage on PopOS 22.04

**Comment by BrunoBernardino** [comment]
I can confirm 3.6.0 also fixed it for me, on a Ubuntu 20.04 variant (Zorin). Server Nextcloud version is 24.0.3. Thanks!

**Comment by hramberger** [comment]
Version 3.6.0 works perfectly fine again on Ubuntu 20.04 for me. It even picked up the the login from the previous working appimage (Nextcloud-3.5.1-x86_64.AppImage). Thanks for the fix!

**Comment by Hudratronium** [comment]
So it seems for at least Ubuntu / Linux clients to be solved.

Sadly nothing new for me, running the windows client 3.6 + NC24.0.5

Every new reconnect to an actual network, i 'll have to reauthenticate / App - Password not working even as the "login" page says successfull, but the app fails to authenticate....

**Comment by zwa-vault** [comment]
Same here, I need to authenticate Windows client 3.6.0 each time I start the client.

**Comment by tgressly** [comment]
Same here, macOS client needs authentication at each launch of the client 3.6.0

**Comment by mophilly** [comment]
Not yet working for me. I have to authenticate on every launch.
MacOS 12.6, NextCloud client 3.6.0 (osx-21.6.0), NextCloud server 24.0.5 running on Koozali SME (CentOS 7)

**Comment by githubtefo** [comment]
In my case I have it stops synchronizing every time I reconnect and I have to reopen the client, very annoying.
Also stable 3.6.0 on Ubutu 22.04 with NextcloudPi as server.

**Comment by RealSerga** [comment]
I have the same problem with appimage version 3.6.4
KDE neon 5.26 
KDE 5.26.5
Linux 5.15

**Comment by Hudratronium** [comment]
@RealSerga 
Hopeless... this one here is closed. should be time to file a new bug report

**Comment by Hudratronium** [comment]
For those who care - new issue open:
https://github.com/nextcloud/desktop/issues/5457

**Comment by serge-vk** [comment]
This was not an issue for me running the AppImage on Ubuntu 24.04 until the last update to 3.14 a few days ago. Now, I have this horrible problem, too. Moreover, after running for a while, NextCloud app freezes and then crashes if I try opening settings or the main dialogue.

**Comment by phi0042** [comment]
> This was not an issue for me running the AppImage on Ubuntu 24.04 until the last update to 3.14 a few days ago. Now, I have this horrible problem, too. Moreover, after running for a while, NextCloud app freezes and then crashes if I try opening settings or the main dialogue.

I have the same annoying problem with the AppImage on Ubuntu 24.04 until the last update to 3.14 a few days ago.

**Comment by phi0042** [comment]
I suggest to use the snap file provided with the Ubuntu software library which contains the version 3.13.4 which does not have the bug.

**Comment by phi0042** [comment]
And the new today's version 3.14.1 does not fix the bug...

**Comment by NielBuys** [comment]
I downgraded my wifes appimage to 3.11.0 that was on my pc. And the problem stopped.

**Comment by rooselle** [comment]
Same problem on Ubuntu 24 since version 3.14. Just downgraded to 3.13 and it works just fine. You can fine the old AppImage here: https://github.com/nextcloud-releases/desktop/releases/tag/v3.13.3 (cc @phi0042)

**Comment by EchedelleLR** [comment]
Is this issue still happening?

**Comment by rooselle** [comment]
I'm currently on 3.15.0 and it works.

**Comment by EchedelleLR** [comment]
I am on 3.15.0, still does not work on Windows.

A reboot = having to relogin.

**Comment by NielBuys** [comment]
I tested on my Ubuntu 24.04 and it did not require authentication after reboot. So it looks like its working.

**Comment by nilsding** [comment]
@EchedelleLR do you have to re-login only once after setting up a new account and a restart of the client, or does that really happen every time?

**Comment by narcisgarcia** [comment]
Remote server: Nextcloud 31
Local environment: Debian GNU/Linux 13 (trixie), MATE desktop, nextcloud-desktop 3.16 (client)

One user profile connects perfectly nextcloud client without having to re-authenticate.
Other user profile has to re-authenticate every time that launches nextcloud client.

What do I need to do so second user's profile keeps authentication saved?!
What can I look to know keyring is working fine?

EDIT: I see keyring was not saving any authentication.

**Comment by camilasan** [comment]
> One user profile connects perfectly nextcloud client without having to re-authenticate. Other user profile has to re-authenticate every time that launches nextcloud client.
> 
> What do I need to do so second user's profile keeps authentication saved?! What can I look to know keyring is working fine?

Are they connected to the same server or it is a different one?

**Comment by narcisgarcia** [comment]
> Are they connected to the same server or it is a different one?

(same server)

I fixed my particular case by fixing keyring storage (with Seahorse application).
I suppose Nextclud-Client assumed a keyring software was saving password (?).
